### DISCUSSION TITLE
Hide and Show livewire  components

### DISCUSSION THREAD

**Discussion by Mohsing511** [livewire/livewire#2075] [4. General]
Hi everyone
i am beginner actually first time using livewire. i need some help.
first: i wants the system in such a way when i click on customer registration button it should trigger "customer-registration-form" (the livewire component) and  "staff-registration-form" should hide. and when i click on staff registration  "customer-registration-form"  should hide (i know i can use jquery but i wants to use something advanced so please guide me
second: i wants to show form only when button is clicked not the the time of loading blade. i am using function for that but i don't know how to show hidden form when the function is triggered.
thanks in advance. here is my code given below
```<div>
    <div class="mt-5 sm:mt-8 sm:flex sm:justify-center lg:justify-center">
        <div class="rounded-md shadow">
            <button wire:click="customerRegistration()" class="w-full flex items-center justify-center px-8 py-3 border border-transparent text-base font-medium rounded-md text-white bg-indigo-600 hover:bg-indigo-700 md:py-4 md:text-lg md:px-10">
                Customer Registration
            </button>
        </div>
        <div class="mt-3 sm:mt-0 sm:ml-3">
            <button wire:click="staffRegistration()" class="w-full flex items-center justify-center px-8 py-3 border border-transparent text-base font-medium rounded-md text-indigo-700 bg-indigo-100 hover:bg-indigo-200 md:py-4 md:text-lg md:px-10">
                Staff Registration
            </button>
        </div>
    </div>
    <div wire:target.hide="customerRegistration">
        @livewire('staff-registration-form')
    </div>
        @livewire('customer-registration-form')

</div>```

**Comment by joshhanley** [comment]
I'd recommend using AlpineJS for this as it was built to work with Livewire to do the interactions you are after https://github.com/alpinejs/alpine.

Give that a go and let us know if you have any questions.

Hope this helps!

**Reply by Mohsing511** [reply]
thanks ok let me try :)
